src/rules/use-selector-in-select.ts

```diff
@@ -34,68 +33,20 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
   },
   defaultOptions: [],
   create: (context) => {
-    let storeName = ''
-    const selectNodes: { node: TSESTree.Node; storeName: string }[] = []
-
-    return {
-      [injectedStore](node: TSESTree.Identifier) {
-        storeName = node.name
-      },
-      [`ClassDeclaration > ClassBody CallExpression[callee.property.name="select"]`](
-        node: TSESTree.CallExpression,
-      ) {
-        const invalidArguments = node.arguments.filter(
-          (arg) => isLiteral(arg) || isArrowFunctionExpression(arg),
-        )
-
-        if (
-          invalidArguments.length &&
-          isMemberExpression(node.callee) &&
-          isMemberExpression(node.callee.object) &&
-          isIdentifier(node.callee.object.property)
-        ) {
-          for (const argument of invalidArguments) {
-            selectNodes.push({
-              node: argument,
-              storeName: node.callee.object.property.name,
-            })
-          }
+    const report = (callNode: TSESTree.CallExpression) => {
+      for (const node of callNode.arguments) {
+        if (isLiteral(node) || isFunctionExpressionLike(node)) {
+          context.report({
+            node,
+            messageId,
+          })
         }
-      },
-      [`ClassDeclaration > ClassBody CallExpression[callee.name="select"]`](
-        node: TSESTree.CallExpression,
-      ) {
-        const invalidArguments = node.arguments.filter(
-          (arg) => isLiteral(arg) || isArrowFunctionExpression(arg),
-        )
+      }
+    }
 
-        if (
-          invalidArguments.length &&
-          node.parent &&
-          isCallExpression(node.parent) &&
-          isMemberExpression(node.parent.callee) &&
-          isMemberExpression(node.parent.callee.object) &&
-          isIdentifier(node.parent.callee.object.property)
-        ) {
-          for (const argument of invalidArguments) {
-            selectNodes.push({
-              node: argument,
-              storeName: node.parent.callee.object.property.name,
-            })
-          }
-        }
-      },
-      ['Program:exit']: () => {
-        if (!storeName) return
-        selectNodes.forEach((n) => {
-          if (n.storeName === storeName) {
-            context.report({
-              node: n.node,
-              messageId,
-            })
-          }
-        })
-      },
+    return {
+      [pipeableSelect(readNgRxStoreNameFromSettings(context.settings))]: report,
+      [storeSelect(readNgRxStoreNameFromSettings(context.settings))]: report,
```

Comment:
The same I mentioned in `select-style`. Also, we could combine these selectors into one using interpolation. I was thinking of something like this (feel free to skip the usage of `filter` + `forEach` if you prefer `for-of` approach):

```suggestion
    const storeName = readNgRxStoreNameFromSettings(context.settings)

    return {
      [`${pipeableSelect(storeName)}, ${storeSelect(storeName)}`]({
        arguments: args,
      }: TSESTree.CallExpression) {
        args
          .filter((node) => isLiteral(node) || isFunctionExpressionLike(node))
          .forEach((node) =>
            context.report({
              node,
              messageId,
            }),
          )
      },
    }
```

Reply:
Oh I keep forgetting this 🤦‍♂️